src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,85 @@
-import React from 'react';
+/* eslint-disable no-console */
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+type Props = {
+  selectedTodo: Todo;
+};
+
+const defaultUser: User = {
+  id: 0,
+  name: 'Default Username',
+  email: 'Default Email',
+  phone: 'Default Phonenumber',
+};
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo }) => {
+  const [user, setUser] = useState<User>(defaultUser);
+  const dispatch = useDispatch();
+
+  const selectedUserId = selectedTodo.userId;
+
+  useEffect(() => {
+    getUser(selectedUserId)
+      .then(setUser)
+      .catch(() => console.error('Error while trying to load user!'));
```

Comment:
Ensure that the `getUser` function correctly fetches user data and handles any potential errors. The error message should provide enough information to debug if the user data fails to load.